src/Repository/QuestionTemplateRepository.php

```diff
@@ -12,15 +13,35 @@ public function getTableName(): string
         return 'question_template';
     }
 
-    public function getNumberOfQuestions(): int
+    public function getNumberOfQuestions(Criteria $criteria): int
     {
-        $sql = 'SELECT count(*) as questionsNumber FROM question_template';
+        $sql = 'SELECT count(*) as questionsNumber FROM question_template ';
+        $sql .= $criteria->toQuerySearch();
         $dbStmt = $this->pdo->prepare($sql);
+        $criteria->bindValueToStatementSearch($dbStmt);
         $dbStmt->execute();
 
         return $dbStmt->fetch(\PDO::FETCH_COLUMN);
     }
 
+    public function findBySearch(Criteria $criteria): array
+    {
+        $sql = 'SELECT * FROM question_template ';
```

Comment:
the repo could have a property table_name or a constant, otherwise if you change the table name you will have to make a lot of changes in the code o

Reply:
also, you can move this to abstract repository if you find a way to get the table name dynamically and BAM, search filter on all entities :D